src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -893,7 +893,11 @@ def _prepare_deployment_properties_unmodified(cmd, deployment_scope, template_fi
         template_obj = _remove_comments_from_json(_urlretrieve(template_uri).decode('utf-8'), file_path=template_uri)
     elif template_spec:
         template_link = TemplateLink(id=template_spec, mode="Incremental")
-        template_obj = show_resource(cmd=cmd, resource_ids=[template_spec]).properties['template']
+        resource = show_resource(cmd=cmd, resource_ids=[template_spec])
+        try:
+            template_obj = resource.properties['mainTemplate']
+        except Exception:  # pylint: disable=broad-except
+            template_obj = resource.properties['template']
```

Comment:
Is this issue caused by breaking change in the new version of `api-version`? If so, could we use `api-version` to distinguish the acquired attributes?